// landingai-react/dist/index.js
"use strict";var W=Object.create;var v=Object.defineProperty;var D=Object.getOwnPropertyDescriptor;var F=Object.getOwnPropertyNames;var Y=Object.getPrototypeOf,q=Object.prototype.hasOwnProperty;var J=(e,o)=>{for(var n in o)v(e,n,{get:o[n],enumerable:!0})},G=(e,o,n,i)=>{if(o&&typeof o=="object"||typeof o=="function")for(let l of F(o))!q.call(e,l)&&l!==n&&v(e,l,{get:()=>o[l],enumerable:!(i=D(o,l))||i.enumerable});return e};var L=(e,o,n)=>(n=e!=null?W(Y(e)):{},G(o||!e||!e.__esModule?v(n,"default",{value:e,enumerable:!0}):n,e)),K=e=>G(v({},"__esModule",{value:!0}),e);var oe={};J(oe,{InferenceContext:()=>U,InferenceResult:()=>R,PhotoCollector:()=>te,initialInferenceContext:()=>O,useInferenceContext:()=>k});module.exports=K(oe);typeof window!="undefined"&&!window.OffscreenCanvas&&(window.OffscreenCanvas=class{constructor(o,n){return this.canvas=document.createElement("canvas"),this.canvas.width=o,this.canvas.height=n,this.canvas.convertToBlob=()=>new Promise(i=>{this.canvas.toBlob(i)}),this.canvas}});var N=require("landingai");var T="c2b33efeeb0a7da04293235a97a603c4606969d3387e85216f367c95af957bdb",Q=`._bottomButton_lf6du_1 {
  margin-top: auto;
  margin-bottom: 8px;
}

._inputGroup_lf6du_6 {
  display: flex;
  flex-direction: column;
  margin-bottom: 0.5rem;
}

._errorLabel_lf6du_12 {
  color: red;
}

._inferenceResult_lf6du_16 {
  position: relative;
  margin-top: 16px;
}

._inferenceSummary_lf6du_21 {
  margin: 16px 0;
  display: flex;
  flex-direction: column;
  padding: 12px;
  gap: 12px;
  background-color: #f4f4f4;
  min-height: 5rem;
  font-size: 15px;
  font-weight: 600;
  border-radius: 0.25rem;
}

._boxAnnotation_lf6du_34 {
  position: absolute;
  border-width: 2px;
  border-style: solid;
}

._segAnnotation_lf6du_40 {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
}

._labelNameCount_lf6du_48 {
  display: flex;
  justify-content: space-between;
  font-weight: 300;
}

._text_lf6du_54 {
  position: absolute;
  font-size: 0.8rem;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 2px 6px;
  border-radius: 4px;
  outline-width: 1px;
  outline-style: solid;
}

._imageContainer_lf6du_66 {
  position: relative;
}

._preview_lf6du_70 {
  border-radius: 0.25rem;
}

._photoPlaceholder_lf6du_74 {
  width: 100%;
  background-color: #f4f4f4;
  padding: 12px;
  padding-bottom: 100%;
  border-radius: 0.25rem;
}

._photoCollectorContainer_lf6du_82 {
  position: relative;
  display: flex;
  flex-direction: column;
}

._photoCollectorButton_lf6du_88 {
  height: 3rem;
  font-size: 1rem;
}

._fileInput_lf6du_93 {
  opacity: 0;
  position: absolute;
  width: 100%;
  height: 100%;
}

._blur_lf6du_100 {
  filter: blur(0.5rem);
}

._ring_lf6du_104 {
  display: inline-block;
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 80px;
  height: 80px;
}

._ring_lf6du_104:after {
  content: " ";
  display: block;
  width: 64px;
  height: 64px;
  margin: 8px;
  border-radius: 50%;
  border: 6px solid #0056fe;
  border-color: #0056fe transparent #0056fe transparent;
  animation: _ring_lf6du_104 1.2s linear infinite;
}

@keyframes _ring_lf6du_104 {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

._labelName_lf6du_48 {
  padding: 0.1rem 0.5rem;
  border-radius: 0.3rem;
  border: 1px solid transparent;
}

._legend_lf6du_141 {
  display: flex;
  flex-wrap: wrap;
}

._legend_lf6du_141 ._labelNameCount_lf6du_48 + ._labelNameCount_lf6du_48 {
  margin-left: 0.5rem;
}`;(function(){if(!(typeof document>"u")&&!document.getElementById(T)){var e=document.createElement("style");e.id=T,e.textContent=Q,document.head.appendChild(e)}})();var s={bottomButton:"_bottomButton_lf6du_1",inputGroup:"_inputGroup_lf6du_6",errorLabel:"_errorLabel_lf6du_12",inferenceResult:"_inferenceResult_lf6du_16",inferenceSummary:"_inferenceSummary_lf6du_21",boxAnnotation:"_boxAnnotation_lf6du_34",segAnnotation:"_segAnnotation_lf6du_40",labelNameCount:"_labelNameCount_lf6du_48",text:"_text_lf6du_54",imageContainer:"_imageContainer_lf6du_66",preview:"_preview_lf6du_70",photoPlaceholder:"_photoPlaceholder_lf6du_74",photoCollectorContainer:"_photoCollectorContainer_lf6du_82",photoCollectorButton:"_photoCollectorButton_lf6du_88",fileInput:"_fileInput_lf6du_93",blur:"_blur_lf6du_100",ring:"_ring_lf6du_104",labelName:"_labelName_lf6du_48",legend:"_legend_lf6du_141"};var t=L(require("react"));var w=require("react"),O={endpoint:"",key:"",secret:""},U=(0,w.createContext)(O),k=()=>(0,w.useContext)(U);var h=require("landingai");var H="ecc1fcf55f61d69588370fbc1b65337fe3c193925dc3efe9488d7bc375b3141d",V=`._bottomButton_lf6du_1 {
  margin-top: auto;
  margin-bottom: 8px;
}

._inputGroup_lf6du_6 {
  display: flex;
  flex-direction: column;
  margin-bottom: 0.5rem;
}

._errorLabel_lf6du_12 {
  color: red;
}

._inferenceResult_lf6du_16 {
  position: relative;
  margin-top: 16px;
}

._inferenceSummary_lf6du_21 {
  margin: 16px 0;
  display: flex;
  flex-direction: column;
  padding: 12px;
  gap: 12px;
  background-color: #f4f4f4;
  min-height: 5rem;
  font-size: 15px;
  font-weight: 600;
  border-radius: 0.25rem;
}

._boxAnnotation_lf6du_34 {
  position: absolute;
  border-width: 2px;
  border-style: solid;
}

._segAnnotation_lf6du_40 {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
}

._labelNameCount_lf6du_48 {
  display: flex;
  justify-content: space-between;
  font-weight: 300;
}

._text_lf6du_54 {
  position: absolute;
  font-size: 0.8rem;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 2px 6px;
  border-radius: 4px;
  outline-width: 1px;
  outline-style: solid;
}

._imageContainer_lf6du_66 {
  position: relative;
}

._preview_lf6du_70 {
  border-radius: 0.25rem;
}

._photoPlaceholder_lf6du_74 {
  width: 100%;
  background-color: #f4f4f4;
  padding: 12px;
  padding-bottom: 100%;
  border-radius: 0.25rem;
}

._photoCollectorContainer_lf6du_82 {
  position: relative;
  display: flex;
  flex-direction: column;
}

._photoCollectorButton_lf6du_88 {
  height: 3rem;
  font-size: 1rem;
}

._fileInput_lf6du_93 {
  opacity: 0;
  position: absolute;
  width: 100%;
  height: 100%;
}

._blur_lf6du_100 {
  filter: blur(0.5rem);
}

._ring_lf6du_104 {
  display: inline-block;
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 80px;
  height: 80px;
}

._ring_lf6du_104:after {
  content: " ";
  display: block;
  width: 64px;
  height: 64px;
  margin: 8px;
  border-radius: 50%;
  border: 6px solid #0056fe;
  border-color: #0056fe transparent #0056fe transparent;
  animation: _ring_lf6du_104 1.2s linear infinite;
}

@keyframes _ring_lf6du_104 {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

._labelName_lf6du_48 {
  padding: 0.1rem 0.5rem;
  border-radius: 0.3rem;
  border: 1px solid transparent;
}

._legend_lf6du_141 {
  display: flex;
  flex-wrap: wrap;
}

._legend_lf6du_141 ._labelNameCount_lf6du_48 + ._labelNameCount_lf6du_48 {
  margin-left: 0.5rem;
}`;(function(){if(!(typeof document>"u")&&!document.getElementById(H)){var e=document.createElement("style");e.id=H,e.textContent=V,document.head.appendChild(e)}})();var C={bottomButton:"_bottomButton_lf6du_1",inputGroup:"_inputGroup_lf6du_6",errorLabel:"_errorLabel_lf6du_12",inferenceResult:"_inferenceResult_lf6du_16",inferenceSummary:"_inferenceSummary_lf6du_21",boxAnnotation:"_boxAnnotation_lf6du_34",segAnnotation:"_segAnnotation_lf6du_40",labelNameCount:"_labelNameCount_lf6du_48",text:"_text_lf6du_54",imageContainer:"_imageContainer_lf6du_66",preview:"_preview_lf6du_70",photoPlaceholder:"_photoPlaceholder_lf6du_74",photoCollectorContainer:"_photoCollectorContainer_lf6du_82",photoCollectorButton:"_photoCollectorButton_lf6du_88",fileInput:"_fileInput_lf6du_93",blur:"_blur_lf6du_100",ring:"_ring_lf6du_104",labelName:"_labelName_lf6du_48",legend:"_legend_lf6du_141"};var d=L(require("react"));function S(e){let{name:o,color:n}=e,i=(0,h.isDark)(n)?"white":"black";return d.default.createElement("span",{style:{backgroundColor:n,color:i,borderColor:i},className:C.labelName},o)}function M(e){let{annotation:o}=e;return o.coordinates?d.default.createElement(X,{...e}):o.bitmap?d.default.createElement(Z,{...e}):d.default.createElement(d.default.Fragment,null)}function X(e){let{annotation:o,imageWidth:n,imageHeight:i,showLabel:l=!1}=e,f=(0,d.useMemo)(()=>{let{coordinates:p,color:_}=o;if(p){let{xmin:m,xmax:c,ymin:x,ymax:b}=p,B=c-m,y=b-x;return{left:`${100*m/n}%`,top:`${100*x/i}%`,width:`${100*B/n}%`,height:`${100*y/i}%`,borderColor:_}}return{}},[o,i,n]),r=(0,d.useMemo)(()=>{let{color:p}=o;return{left:0,top:-4,backgroundColor:p,color:(0,h.isDark)(p)?"white":"black",outlineColor:(0,h.isDark)(p)?"white":"black",transform:"translateY(-100%)"}},[o,i,n]);return d.default.createElement("div",{className:C.boxAnnotation,style:f},l&&d.default.createElement("div",{className:C.text,style:r},o.name))}function Z(e){let{annotation:o,imageWidth:n,imageHeight:i}=e,{bitmap:l,color:f}=o;if(!l)return null;let[r,p]=(0,d.useState)();return(0,d.useEffect)(()=>{let _;return(0,h.convertCompressedBitMapToBlob)(l,f,n,i).then(m=>{_=URL.createObjectURL(m),p(_)}),()=>{_&&URL.revokeObjectURL(_)}},[]),r?d.default.createElement("img",{className:C.segAnnotation,src:r}):null}var R=e=>{let{image:o,showLabels:n=!1,onPredictError:i}=e,l=k(),f=(0,t.useRef)(null),[r,p]=(0,t.useState)(),_=(0,t.useMemo)(()=>(0,N.predictionsToAnnotations)(r),[r]),m=(0,t.useMemo)(()=>{var a,u;return(u=(a=r==null?void 0:r.predictions)==null?void 0:a.labelName)!=null?u:""},[r]),[c,x]=(0,t.useState)(!1),[b,B]=(0,t.useState)(),y=(0,t.useMemo)(()=>{let a=_.reduce((u,g)=>{var j,z;return{...u,[g.name]:{name:g.name,color:g.color,count:((z=(j=u[g.name])==null?void 0:j.count)!=null?z:0)+1}}},{});return Object.values(a)},[_]),E=(0,t.useCallback)(async a=>{try{x(!0);let u=await(0,N.getInferenceResult)(l,a);p(u)}catch(u){i==null||i(u)}finally{x(!1)}},[l]);return(0,t.useEffect)(()=>{if(o){let a=URL.createObjectURL(o);return B(a),()=>URL.revokeObjectURL(a)}},[o]),(0,t.useEffect)(()=>{o&&E(o)},[o,E]),t.default.createElement(t.default.Fragment,null,t.default.createElement("div",{className:s.inferenceResult,"data-testid":"inference-result"},!b&&t.default.createElement("div",{className:s.photoPlaceholder},"Your image will be displayed here."),b&&t.default.createElement("div",{className:s.imageContainer},t.default.createElement("img",{className:`${s.preview} ${c?s.blur:""}`,ref:f,src:b,alt:"result",width:"100%",style:{verticalAlign:"middle"}}),f.current&&!c&&_.map(a=>t.default.createElement(M,{key:a.id,annotation:a,imageWidth:f.current.naturalWidth,imageHeight:f.current.naturalHeight,showLabel:n})),c&&t.default.createElement("div",{className:s.ring})),b&&!c&&!!r&&t.default.createElement("div",{className:s.inferenceSummary},(r==null?void 0:r.type)==="ClassificationPrediction"&&!r.backbonetype&&t.default.createElement("div",null,"Class: ",m),r.backbonetype==="ObjectDetectionPrediction"&&t.default.createElement(t.default.Fragment,null,t.default.createElement("div",null,"Total: ",_.length," objects detected"),y.map(({name:a,count:u,color:g})=>t.default.createElement("div",{className:s.labelNameCount,key:a},t.default.createElement("span",null,"Number of ",t.default.createElement(S,{name:a,color:g})),t.default.createElement("span",null,u)))),((r==null?void 0:r.type)==="SegmentationPrediction"||r.backbonetype==="SegmentationPrediction")&&t.default.createElement(t.default.Fragment,null,t.default.createElement("div",null,"Legend"),t.default.createElement("div",{className:s.legend},y.map(({name:a,color:u})=>t.default.createElement("div",{className:s.labelNameCount,key:a},t.default.createElement("span",null,t.default.createElement(S,{name:a,color:u})))))))))};var A=L(require("react"));var $="4039e5cd3cba6bbc3d309eb0abe8f06aedffc8892935a6883e152080024f593d",ee=`._bottomButton_lf6du_1 {
  margin-top: auto;
  margin-bottom: 8px;
}

._inputGroup_lf6du_6 {
  display: flex;
  flex-direction: column;
  margin-bottom: 0.5rem;
}

._errorLabel_lf6du_12 {
  color: red;
}

._inferenceResult_lf6du_16 {
  position: relative;
  margin-top: 16px;
}

._inferenceSummary_lf6du_21 {
  margin: 16px 0;
  display: flex;
  flex-direction: column;
  padding: 12px;
  gap: 12px;
  background-color: #f4f4f4;
  min-height: 5rem;
  font-size: 15px;
  font-weight: 600;
  border-radius: 0.25rem;
}

._boxAnnotation_lf6du_34 {
  position: absolute;
  border-width: 2px;
  border-style: solid;
}

._segAnnotation_lf6du_40 {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
}

._labelNameCount_lf6du_48 {
  display: flex;
  justify-content: space-between;
  font-weight: 300;
}

._text_lf6du_54 {
  position: absolute;
  font-size: 0.8rem;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 2px 6px;
  border-radius: 4px;
  outline-width: 1px;
  outline-style: solid;
}

._imageContainer_lf6du_66 {
  position: relative;
}

._preview_lf6du_70 {
  border-radius: 0.25rem;
}

._photoPlaceholder_lf6du_74 {
  width: 100%;
  background-color: #f4f4f4;
  padding: 12px;
  padding-bottom: 100%;
  border-radius: 0.25rem;
}

._photoCollectorContainer_lf6du_82 {
  position: relative;
  display: flex;
  flex-direction: column;
}

._photoCollectorButton_lf6du_88 {
  height: 3rem;
  font-size: 1rem;
}

._fileInput_lf6du_93 {
  opacity: 0;
  position: absolute;
  width: 100%;
  height: 100%;
}

._blur_lf6du_100 {
  filter: blur(0.5rem);
}

._ring_lf6du_104 {
  display: inline-block;
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 80px;
  height: 80px;
}

._ring_lf6du_104:after {
  content: " ";
  display: block;
  width: 64px;
  height: 64px;
  margin: 8px;
  border-radius: 50%;
  border: 6px solid #0056fe;
  border-color: #0056fe transparent #0056fe transparent;
  animation: _ring_lf6du_104 1.2s linear infinite;
}

@keyframes _ring_lf6du_104 {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

._labelName_lf6du_48 {
  padding: 0.1rem 0.5rem;
  border-radius: 0.3rem;
  border: 1px solid transparent;
}

._legend_lf6du_141 {
  display: flex;
  flex-wrap: wrap;
}

._legend_lf6du_141 ._labelNameCount_lf6du_48 + ._labelNameCount_lf6du_48 {
  margin-left: 0.5rem;
}`;(function(){if(!(typeof document>"u")&&!document.getElementById($)){var e=document.createElement("style");e.id=$,e.textContent=ee,document.head.appendChild(e)}})();var I={bottomButton:"_bottomButton_lf6du_1",inputGroup:"_inputGroup_lf6du_6",errorLabel:"_errorLabel_lf6du_12",inferenceResult:"_inferenceResult_lf6du_16",inferenceSummary:"_inferenceSummary_lf6du_21",boxAnnotation:"_boxAnnotation_lf6du_34",segAnnotation:"_segAnnotation_lf6du_40",labelNameCount:"_labelNameCount_lf6du_48",text:"_text_lf6du_54",imageContainer:"_imageContainer_lf6du_66",preview:"_preview_lf6du_70",photoPlaceholder:"_photoPlaceholder_lf6du_74",photoCollectorContainer:"_photoCollectorContainer_lf6du_82",photoCollectorButton:"_photoCollectorButton_lf6du_88",fileInput:"_fileInput_lf6du_93",blur:"_blur_lf6du_100",ring:"_ring_lf6du_104",labelName:"_labelName_lf6du_48",legend:"_legend_lf6du_141"};var P=require("image-conversion"),te=e=>{let{setImage:o}=e,n=async i=>{let l=i.target.files;if(l!=null&&l.length&&l[0]){let f=await(0,P.compress)(l[0],{size:1e3,type:P.EImageType.JPEG,quality:.9});o(f)}};return A.default.createElement("div",{className:I.photoCollectorContainer,"aria-label":"Select a photo"},A.default.createElement("button",{className:I.photoCollectorButton},"Select a photo"),A.default.createElement("input",{onChange:n,className:I.fileInput,type:"file",accept:"image/*;capture=camera","data-testid":"select-photo-input"}))};0&&(module.exports={InferenceContext,InferenceResult,PhotoCollector,initialInferenceContext,useInferenceContext});
